File: Java/labs/lab20_State/task_2_2/src/MediaPlayer.java
import java.util.ArrayList;
import java.util.List;

public class MediaPlayer {

    final private List<String> tracks = new ArrayList<>();
    private State state;
    private int currentTrackNum = 0;


    public String getCurrentTrack() {
        return tracks.get(currentTrackNum);
    }

    public void setTrackNum(int trackNum) {
        if (trackNum < 0 || trackNum >= tracks.size()) {
            return;
        }

        this.currentTrackNum = trackNum;
    }

    public int getCurrentTrackNum() {
        return currentTrackNum;
    }

    public List<String> getTracks() {
        return tracks;
    }

    public void addTrack(String track) {
        tracks.add(track);
    }

    public void setState(State state) {
        this.state = state;
    }

    public State getState() {
        return state;
    }

    public void play() {

    }

    public void pause() {

    }

    public void next() {

    }

    public void prev() {

    }

    public void stop() {

    }

}

File: Cpp/lab20_State/task_3_1/src/media_player.cpp
#include "media_player.hpp"


MediaPlayer::~MediaPlayer()
{
  if (this->state != nullptr)
  {
    delete this->state;
    this->state = nullptr;
  }
}

std::string const& MediaPlayer::getCurrentTrack() const
{
  return this->tracks.at(this->currentTrackNum);
}

void MediaPlayer::setTrackNum(size_t trackNum)
{
  if (trackNum >= this->tracks.size())
  {
    return;
  }

  this->currentTrackNum = trackNum;
}

size_t MediaPlayer::getCurrentTrackNum() const
{
  return this->currentTrackNum;
}

std::vector<std::string> const& MediaPlayer::getTracks() const
{
  return this->tracks;
}

void MediaPlayer::addTrack(std::string const& track)
{
  this->tracks.push_back(track);
}

void MediaPlayer::setState(State* state)
{
  if (this->state != nullptr)
  {
    delete this->state;
  }

  this->state = state;
}

State* MediaPlayer::getState()
{
  return this->state;
}

void MediaPlayer::play()
{

}

void MediaPlayer::pause()
{

}

void MediaPlayer::next()
{

}

void MediaPlayer::prev()
{

}

void MediaPlayer::stop()
{

}
